File: quan_ly_tu_do_API_one/quan_ly_tu_do_API/Models/HealthLog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using System.Text.Json.Serialization;
namespace quan_ly_tu_do_API.Models;

[Table("health_logs")]
[Index("UserId", "LogDate", Name = "idx_health_logs_user_date", IsDescending = new[] { false, true })]
[Index("UserId", "LogDate", Name = "uq_health_user_date", IsUnique = true)]
public partial class HealthLog
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; }

    [Column("user_id")]
    public Guid UserId { get; set; }

    [Column("log_date")]
    public DateOnly LogDate { get; set; }

    [Column("heart_rate")]
    [StringLength(20)]
    public string? HeartRate { get; set; }

    [Column("blood_pressure")]
    [StringLength(20)]
    public string? BloodPressure { get; set; }

    [Column("weight", TypeName = "decimal(5, 2)")]
    public decimal? Weight { get; set; }

    [Column("sleep_hours", TypeName = "decimal(4, 2)")]
    public decimal? SleepHours { get; set; }

    [Column("notes")]
    public string? Notes { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [ForeignKey("UserId")]
    [InverseProperty("HealthLogs")]
    [JsonIgnore]
    public virtual User? User { get; set; }
}

File: tuquanaoAI/lib/models/health_log.dart
class HealthLog {
  final String id;
  final String userId;
  final String logDate;
  final String heartRate;
  final String bloodPressure;
  final String weight;
  final String sleep;
  final String? notes;

  const HealthLog({
    required this.id,
    required this.userId,
    required this.logDate,
    this.heartRate = '',
    this.bloodPressure = '',
    this.weight = '',
    this.sleep = '',
    this.notes,
  });

  factory HealthLog.fromJson(Map<String, dynamic> j) => HealthLog(
    id: j['id']?.toString() ?? '',
    userId: j['userId']?.toString() ?? '',
    logDate: j['logDate']?.toString() ?? '',
    heartRate: j['heartRate']?.toString() ?? '',
    bloodPressure: j['bloodPressure']?.toString() ?? '',
    weight: j['weight']?.toString() ?? '',
    sleep: j['sleepHours']?.toString() ?? '',
    notes: j['notes'],
  );

  Map<String, dynamic> toJson(String userId) => {
    'userId': userId,
    'logDate': logDate,
    'heartRate': heartRate.isEmpty ? null : heartRate,
    'bloodPressure': bloodPressure.isEmpty ? null : bloodPressure,
    'weight': weight.isEmpty ? null : double.tryParse(weight),
    'sleepHours': sleep.isEmpty ? null : double.tryParse(sleep),
    'notes': notes,
  };
}